### DISCUSSION TITLE
Django package maintainer working group

### DISCUSSION THREAD

**Discussion by tim-schilling** [django-commons/membership#39] [General]
In case folks haven't heard, but there's a working group proposed to support package maintainers in the Django community. It's still in a review state, so please give it a read and provide your feedback: https://github.com/django/dsf-working-groups/pull/19

**Comment by ulgens** [comment]
@tim-schilling I tried to catch up with the discussions, but I couldn't really understand the plan/expectation about the overlap between that new working group's and Django Commons' responsibilities.

**Reply by tim-schilling** [reply]
I think it's best to focus on the goals of that working group. Any Working Group is a delegation of power/responsibility from the DSF board.

Given the goals of the WG:

> - To facilitate discussions and share knowledge about common issues or tasks in package maintenance.
> - To coordinate efforts to improve the health and quality of packages in the Django ecosystem.
> - To advocate for the needs of package maintainers in the Django community.

I believe it's priority is to improve communication within the Django community and to define standards or guidelines that package maintainers could aim for. Or at least use as a reference. I suspect it's actual tasks are going to be largely defined by the people who are in the working group.

Does the above help your understanding? If so, would you be interested in helping out on the WG? I think right now it really needs someone to push those discussions forward and suggest concrete changes based on the existing discussions.

I'd love to see that WG get pushed forward and get off the ground. It'd make some of what Django Commons is attempting to do easier. For example, if the WG were to set "Hey, this is a good way to publishing a package and how to run a test matrix in an environmently friendly manner" Django Commons can actually implement those guidelines and find ways to reduce the amount of custodian tasks maintainers do.

**Reply by ulgens** [reply]
> I suspect it's actual tasks are going to be largely defined by the people who are in the working group.

This makes perfect sense.

> Does the above help your understanding? If so, would you be interested in helping out on the WG? 

Yes, it does. Thanks a lot. I still can't say I fully understand the expectations from the proposed WG, but I see the goal, and I'd like to help achieve it. I'll check the discussion under the proposal.

> "Hey, this is a good way to publishing a package and how to run a test matrix in an environmently friendly manner."

This would be my dream scenario of contributing back to the community 😄 All those once useful but now broken packages, wasted time and effort, and all that frustration when you find something useful but then realize it's not maintained anymore... There are lots of common problems and common solutions that can be applied to them, but they require proper research, testing, organization, and communication.  That would be a worthy challenge.
